python/tank_vendor/shotgun_authentication/interactive_authentication.py

```diff
@@ -0,0 +1,432 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+UI and console based login for Toolkit.
+"""
+
+# Using "with" with the lock to make sure it is always released.
+
+from __future__ import with_statement
+from getpass import getpass
+import threading
+from .errors import AuthenticationError, AuthenticationDisabled, AuthenticationCancelled
+from . import session_cache
+
+
+# FIXME: Quick hack to easily disable logging in this module while keeping the
```

Comment:
While this code is not obvious to turn on, I would not remove the logging that is present. It has a lot of value and I use it all the time when I am developing the authentication feature. We'd be losing all the work I put in making it useful to debug it. At best I can comment the lines where I log stuff.